controllers/nutanixcluster_controller.go

```diff
@@ -272,6 +278,25 @@ func (r *NutanixClusterReconciler) reconcileNormal(rctx *nctx.ClusterContext) (r
 	return reconcile.Result{}, nil
 }
 
+func (r *NutanixClusterReconciler) reconcileFailureDomains(rctx *nctx.ClusterContext) error {
+	log := ctrl.LoggerFrom(rctx.Context)
+	if len(rctx.NutanixCluster.Spec.FailureDomains) == 0 {
+		log.V(1).Info("no failure domains defined on cluster")
+		conditions.MarkTrue(rctx.NutanixCluster, infrav1.NoFailureDomainsReconciled)
+		return nil
+	}
+	log.V(1).Info("Reconciling failure domains for cluster")
+	// If failure domains is nil on status object, first create empty slice
+	if rctx.NutanixCluster.Status.FailureDomains == nil {
+		rctx.NutanixCluster.Status.FailureDomains = make(capiv1.FailureDomains, 0)
+	}
+	for _, fd := range rctx.NutanixCluster.Spec.FailureDomains {
+		rctx.NutanixCluster.Status.FailureDomains[fd.Name] = capiv1.FailureDomainSpec{ControlPlane: fd.ControlPlane}
```

Comment:
can we document and point to some capi code where it takes care of handling of failureDOmains just by setting it on the status field?

Reply:
Updated description of the PR with the link.